File: extensions/shared-youtube/library/src/main/java/app/argusos/extension/shared/patches/DisableQUICProtocolPatch.java
package app.argusos.extension.shared.patches;

import app.argusos.extension.shared.settings.SharedYouTubeSettings;

@SuppressWarnings("unused")
public class DisableQUICProtocolPatch {

    public static boolean disableQUICProtocol(boolean original) {
        boolean isDisabled = SharedYouTubeSettings.DISABLE_QUIC_PROTOCOL.get();
        return !isDisabled && original;
    }
}

File: patches/src/main/kotlin/app/argusos/patches/music/misc/quic/DisableQUICProtocolPatch.kt
package app.argusos.patches.music.misc.quic

import app.argusos.patches.music.misc.extension.sharedExtensionPatch
import app.argusos.patches.music.misc.settings.PreferenceScreen
import app.argusos.patches.music.misc.settings.settingsPatch
import app.argusos.patches.music.shared.Constants.COMPATIBILITY_YOUTUBE_MUSIC
import app.argusos.patches.shared.misc.quic.disableQUICProtocolPatch

@Suppress("unused")
val disableQUICProtocolPatchMusic = disableQUICProtocolPatch(
    block = {
        dependsOn(
            sharedExtensionPatch,
            settingsPatch
        )

        compatibleWith(COMPATIBILITY_YOUTUBE_MUSIC)
    },
    preferenceScreen = PreferenceScreen.MISC
)